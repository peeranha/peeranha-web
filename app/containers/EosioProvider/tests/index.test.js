import React from 'react';
import { EosioProvider, mapDispatchToProps } from '../index';

const child = <div>children</div>;
React.Children.only = jest.fn().mockImplementation(() => child);

const cmp = new EosioProvider();
cmp.props = {
  initializing: false,
  initEosio: jest.fn(),
};

describe('<EosioProvider />', () => {
  it('render, @initializing is false', () => {
    cmp.props.initializing = false;
    expect(cmp.render()).toMatchSnapshot();
  });

  it('render, @initializing is true', () => {
    cmp.props.initializing = true;
    expect(cmp.render()).toMatchSnapshot();
  });

  it('mapDispatchToProps test', () => {
    const test = 'test';
    const dispatch = () => test;

    expect(typeof mapDispatchToProps(dispatch) === 'object').toBe(true);
    expect(mapDispatchToProps(dispatch).dispatch).toBe(dispatch);
    expect(mapDispatchToProps(dispatch).initEosio()).toBe(test);
  });

  it('componentDidMount', () => {
    cmp.componentDidMount();
    expect(cmp.props.initEosio).toHaveBeenCalled();
  });
});
