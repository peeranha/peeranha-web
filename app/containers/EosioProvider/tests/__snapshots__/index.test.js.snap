// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<EosioProvider /> render, @initializing is false 1`] = `
<div>
  <div>
    children
  </div>
</div>
`;

exports[`<EosioProvider /> render, @initializing is true 1`] = `
<div>
  <Wrapper>
    <LoadingIndicator />
  </Wrapper>
</div>
`;
